runtime-light/stdlib/server/http-functions.cpp

```diff
@@ -181,3 +181,16 @@ void f$setrawcookie(const string &name, const string &value, int64_t expire_or_o
   }
   kphp::http::header({static_SB_spare.c_str(), static_SB_spare.size()}, false, kphp::http::status::NO_STATUS);
 }
+
+array<string> f$headers_list() noexcept {
+  auto &http_server_instance_st{HttpServerInstanceState::get()};
+  const auto &headers{http_server_instance_st.headers()};
+
+  array<string> list;
+  for (const auto & header : headers) {
+    string value{(header.first + ": " + header.second).c_str()};
```

Comment:
No need to temporary `std::string` here. It's better to use a combination of `reserve_at_least` and `append`